src/main.rs

```diff
@@ -288,24 +288,13 @@ pub fn set_var(variables: &mut Variables, name: &str, value: &str) {
     }
 }
 
-fn print_prompt(modes: &Vec<Mode>) {
-    for mode in modes.iter().rev() {
-        if mode.value {
-            print!("+ ");
-        } else {
-            print!("- ");
-        }
-    }
+fn print_prompt(modes: &[Mode]) {
+    let mode = modes.iter().rev().fold(String::new(), |acc, mode| {
```

Comment:
Perhaps naming the string that we print out `mode` is confusing `mode` is also used in the fold. I am thinking something like `mode_prompt` or `prompt_prefix` would be more descriptive.